pkg/gardenlet/controller/seed/seed_lease_control.go

```diff
@@ -0,0 +1,135 @@
+	if err := c.checkSeedConnection(seed); err != nil {
+		return fmt.Errorf("[SEED LEASE] cannot establish connection with Seed %s: %v", key, err)
+	}
+
+	condition := helper.GetCondition(seed.Status.Conditions, gardencorev1beta1.SeedGardenletReady)
+	if condition == nil {
+		err := c.initGardenletReadyCondition(seed)
+		if err != nil {
+			return err
+		}
+	}
+
+	clock := clockpkg.RealClock{}
+	seedOwnerReference := buildSeedOwnerReference(seed)
+	leaseControl := lease.NewLeaseController(clock, c.k8sGardenClient.Kubernetes(), leaseResyncSeconds, seed.GetName(), gardenercorev1betaconstants.GardenSeedLeaseNamespace, *seedOwnerReference)
+	if err := leaseControl.Sync(); err != nil {
+		return err
+	}
+
+	if condition.Status == "Unknown" {
+		err := c.updateGardenletReadyCondition(seed)
+		if err != nil {
+			return err
+		}
+	}
+
+	durationAfter := time.Duration(leaseResyncSeconds * time.Second)
+	c.seedLeaseQueue.AddAfter(key, durationAfter)
+	return nil
+}
+
+func buildSeedOwnerReference(seed *gardencorev1beta1.Seed) *metav1.OwnerReference {
+	return &metav1.OwnerReference{
+		APIVersion: gardencorev1beta1.SchemeGroupVersion.WithKind("Seed").Version,
+		Kind:       gardencorev1beta1.SchemeGroupVersion.WithKind("Seed").Kind,
+		Name:       seed.GetName(),
+		UID:        seed.GetUID(),
+	}
+}
+
+func (c *Controller) initGardenletReadyCondition(seed *gardencorev1beta1.Seed) error {
+	conditionGardenletReady := helper.InitCondition(gardencorev1beta1.SeedGardenletReady)
+	conditionGardenletReady = helper.UpdatedCondition(conditionGardenletReady, gardencorev1beta1.ConditionTrue, "GardenletReady", "Gardenlet is posting ready status.")
+	return c.updateSeedStatus(seed, conditionGardenletReady)
+}
+
+func (c *Controller) updateGardenletReadyCondition(seed *gardencorev1beta1.Seed) error {
+	var conditionGardenletReady = helper.GetOrInitCondition(seed.Status.Conditions, gardencorev1beta1.SeedGardenletReady)
+	conditionGardenletReady = helper.UpdatedCondition(conditionGardenletReady, gardencorev1beta1.ConditionTrue, "GardenletReady", "Gardenlet is posting ready status.")
+	return c.updateSeedStatus(seed, conditionGardenletReady)
+}
+
+func (c *Controller) updateSeedStatus(seed *gardencorev1beta1.Seed, conditionGardenletReady gardencorev1beta1.Condition) error {
+	_, err := kutil.TryUpdateSeedConditions(c.k8sGardenClient.GardenCore(), retry.DefaultBackoff, seed.ObjectMeta,
+		func(seed *gardencorev1beta1.Seed) (*gardencorev1beta1.Seed, error) {
+			seed.Status.Conditions = helper.MergeConditions(seed.Status.Conditions, conditionGardenletReady)
+			return seed, nil
+		},
+	)
+	return err
+}
+
+func (c *Controller) checkSeedConnection(seed *gardencorev1beta1.Seed) error {
+	client, err := seedpkg.GetSeedClient(context.TODO(), c.k8sGardenClient.Client(), c.config.SeedClientConnection.ClientConnectionConfiguration, c.config.SeedSelector == nil, seed.Name)
+	if err != nil {
+		return err
+	}
+
+	_, err = client.Kubernetes().CoreV1().Pods("").List(v1.ListOptions{})
```

Comment:
Why do we list the pods? Can we just call `/healthz` endpoint of the API server?